web-registry/src/pages/ProjectMetadata/hooks/useProjectMetadataSubmit.tsx

```diff
@@ -0,0 +1,49 @@
+import { useCallback } from 'react';
+import { merge } from 'lodash';
+import { cloneDeep } from 'lodash';
+import { ProjectMetadataValues } from '../../../components/organisms';
+import { ProjectByIdQuery } from '../../../generated/graphql';
+import { useUpdateProjectByIdMutation } from '../../../generated/graphql';
+
+type Props = {
+  project?: ProjectByIdQuery['projectById'];
+  projectId?: string;
+  updateProject: ReturnType<typeof useUpdateProjectByIdMutation>[0];
+};
+
+type ReturnedType = (values: ProjectMetadataValues) => Promise<void>;
+
+const useProjectMetadataSubmit = ({
+  project,
+  projectId,
+  updateProject,
+}: Props): ReturnedType => {
+  const projectMetadataSubmit = useCallback(
+    async function submit(values: ProjectMetadataValues): Promise<void> {
+      const parsedMetaData = JSON.parse(values.metadata);
+      const projectMetadata = cloneDeep(project?.metadata);
+      merge(projectMetadata, parsedMetaData);
```

Comment:
We're extending the current metadata with new values.
The IRI generation will be done at the review step.